vac/raw/mix.md

```diff
@@ -0,0 +1,659 @@
+as Proof of Work and delayed forwarding address common attacks on anonymity
+networks, such as spam and timing analysis.
+
+By incorporating these features, the Mix protocol aims to provide a robust
+anonymity layer within the libp2p ecosystem, enabling developers to easily
+incorporate privacy features into their applications.
+
+## Specification
+
+### 1. Protocol Identifier
+
+The Mix protocol is identified by the string `"/mix/1.0.0"`.
+
+### 2. Custom Mix Protocol
+
+The Mix protocol is designed as a standalone protocol, 
+identified by the protocol identifier `"/mix/1.0.0"`. 
+This approach allows the Mix protocol to operate independently,
+decoupled from specific applications, 
+providing greater flexibility and reusability across various libp2p protocols. 
+By doing so, the Mix protocol can evolve independently, 
+focusing on its core functionality without being tied to the development and maintenance cycles of other protocols.
+
+#### 2.1 Mix Nodes Roles
+
+All nodes participating in the Mix protocol are considered as mix nodes. They
+have the capability to create/process and forward Sphinx packets. Mix nodes can
+play different roles depending on their position in a particular message path:
+
+- **Sender Node**
+  - A mix node that initiates the anonymous message publishing process.
+  - Responsible for:
+    - Path selection.
+    - Sphinx packet creation.
+    - Initiating the message routing through the mix network.
+  - Must run both the Mix protocol instance and the instance of the libp2p
+    protocol for the message being published (_e.g.,_ GossipSub, Ping, etc.).
+- **Intermediary Mix Node**
+  - A mix node that is neither the sender nor the exit node in a message path.
+  - Responsible for:
+    - Receiving Sphinx packets.
+    - Processing (decrypting and re-encrypting) Sphinx packets.
+    - Forwarding processed packets to the next node in the path.
+  - Only needs to run the Mix protocol instance.
+- **Exit Node**
+  - The final mix node in a message path.
+  - Responsible for:
+    - Receiving and processing the final Sphinx packet.
+    - Extracting the original message.
+    - Disseminating the decrypted message using the appropriate libp2p protocol.
+  - Must run both the Mix protocol instance and the instance of the libp2p
+    protocol for the message being published.
+
+#### 2.2 Roles Flexibility
+
+A single mix node can play different roles in different paths:
+
+- It can be a sender node for messages it initiates.
+- It can be an intermediary node for messages it is forwarding.
+- It can be an exit node for messages it is disseminating.
+
+#### 2.3 Incentives
+
+To publish an anonymous libp2p message (_e.g.,_ GossipSub, Ping, etc.), nodes
+MUST run a mix node instance. This requirement serves as an incentive for nodes
+to participate in the mix network, as it allows them to benefit from the
+anonymity features while also contributing to the network's overall anonymity
+and robustness.
```

Comment:
In case there is no enough mixnodes in a network, may it cause some anonymity leakages for mixnodes? If so, is there any theroetical threshold percentage for being safe?

Reply:
You raise an important point. Indeed, having too few mix nodes can potentially lead to anonymity leakages, and this is not exclusive to our mix protocol but a general concern for all mix networks. While there's no universally agreed-upon threshold, the general principle is that more mix nodes provide better anonymity. 

To mitigate risks in smaller networks, we could consider (in the later versions):
1. Implementing stricter node selection criteria
2. Introducing additional cover traffic

We should keep this in mind as we develop and deploy the protocol.